Use anchor tag in a11y link button

maxichrome opened the pull request:
Removes a layer of nesting that was kinda wonky — though it does look like it messes up the clean animation. could be worth looking into if that's a big deal but idk how much of a priority it should be...


tagging @jamiepine @Brendonovich @ryanhaticus to get in on this since it's in a 3p repo

ryanhaticus commented:
While I agree with the issue, I think an alternative "LinkButton" might be a better alternative. Concerned about the `@ts-expect-error` and an ambiguous definition of the Button element.

maxichrome commented:
Hmm, you may have a point with that (winning points for explicitness) but I think overall the simplicity here offers a better DX. updated with changes from the upstream `main` branch and streamlined some of the code in the Button component -- notably, removed the ts-expect-error
